shtab/__init__.py

```diff
@@ -145,13 +137,25 @@ def get_bash_commands(root_parser, root_prefix, choice_functions=None):
             # `add_argument('subcommand', choices=shtab.Required.FILE)`)
             _{root_parser.prog}_{subcommand}_COMPGEN=_shtab_compgen_files
     """
-    choice_type2fn = dict(CHOICE_FUNCTIONS_BASH)
+    choice_type2fn = {k: v["bash"] for k, v in CHOICE_FUNCTIONS.items()}
     if choice_functions:
         choice_type2fn.update(choice_functions)
 
     fd = io.StringIO()
     root_options = []
 
+    def get_optional_actions(parser):
+        """Flattened list of all `parser`'s optional actions."""
+        return sum(
+            (
+                # TODO: if hasattr(opt, "complete")?
```

Comment:
Could create an issue instead of TODO, but your call.

Reply:
👍 if we keep TODO it should include a link to the proper ticket at least. And ticket should link to it.

Reply:
this TODO was never implemented (even in DVC's original scripts).

- add_argument("positional").complete = shtab.FILE
  + zsh: works
  + bash: didn't work in DVC, works now with shtab
- add_argument("--optional").complete = shtab.FILE
  + zsh: works
  + bash: never worked

Could open an issue...

Reply:
```suggestion
```

Reply:
opened #11